orquesta/conducting.py

```diff
@@ -653,8 +653,12 @@ def update_task_state(self, task_id, route, event):
         task_state_idx = self._get_task_state_idx(task_id, route)
 
         # If task is already completed and in cycle, then create new task state entry.
-        if (self.graph.in_cycle(task_id) and
-                task_state_entry.get('status') in statuses.COMPLETED_STATUSES):
+        # Unfortunately, the method in the graph to check for cycle is too simple and
+        # misses forks that extends from the cycle. The check here assumes that the
+        # last task entry is already completed and the new task status is one of the
+        # starting statuses, then there is high likelihood that this is a cycle.
+        if (task_state_entry.get('status') in statuses.COMPLETED_STATUSES and
```

Comment:
Above line 655 commented `If task is already completed and in cycle`, the code here did not check if task is in cycle `self.graph.in_cycle(task_id)`.

Reply:
The comment `If task is already completed and in cycle` is saying what must be checked.  But if you look at the description right after that, because the method in `networkx` library to check cycle is too simple, here is what we are doing.

Reply:
Should change comment to remove cycle?

Reply:
No, we shouldn't change comment to remove "cycle". We want to detect whether the current task is part of a cycle.   It is just that the method in networkx is unable to determine that this very specific use case with a fork stems from a cycle.